File: spaceInvaders/src/main/java/spaceInvaders/AlienNave.java
package spaceInvaders;

import javafx.scene.image.ImageView;
import java.util.Random;

public class AlienNave extends Nave {
    private final Random random = new Random();

    public AlienNave(String spritePath, double x, double y) {
        setPosX(x);
        setPosY(y);
        setSprite(spritePath);
        setSpriteView(new ImageView(getSprite()));

        // Posiciona o sprite na cena
        getSpriteView().setX(getPosX());
        getSpriteView().setY(getPosY());

        setVelocidade(1);
        setDescida(20);
        setMovimentoParaDireita(true);
    }

    public void moverHorizontal(boolean paraDireita, double velocidade) {
        double x = getSpriteView().getX();
        if (paraDireita) {
            x += velocidade;
        } else {
            x -= velocidade;
        }
        getSpriteView().setX(x);
        setPosX(x);
    }

    public void descer() {
        double y = getSpriteView().getY();
        double novaY = y + getDescida();
        getSpriteView().setY(novaY);
        setPosY(novaY);
    }

    public Disparo atacar() {
        // Centraliza o disparo horizontalmente em relação à nave
        double disparoX = getPosX() + getSpriteView().getFitWidth() / 2 - 5;
        double disparoY = getPosY() + getSpriteView().getFitHeight();

        return new Disparo(disparoX, disparoY, false); //false = o disparo não veio do player
    }

    @Override
    public void destruir() {
        getSpriteView().setVisible(false);
    }
}

File: SpaceInvaders(Python)/src/AlienNave.py
# -*- coding: utf-8 -*-
from Nave import Nave
from Disparo import Disparo


class AlienNave(Nave):
    """Nave inimiga."""

    def __init__(self, sprite_path: str, x: int, y: int):
        super().__init__(sprite_path, x, y, hp=1)

    def mover_horizontal(self, direita: bool, velocidade: float):
        """Move a nave alienígena horizontalmente."""
        novo_x = self.get_pos_x() + (velocidade if direita else -velocidade)
        self.set_pos_x(novo_x)

    def descer(self, pixels: int = 20):
        """Faz a nave alienígena descer na tela."""
        self.set_pos_y(self.get_pos_y() + pixels)

    def atacar(self):
        """Cria um disparo partindo da nave alienígena."""
        disparo_x = self.get_pos_x() + self.get_sprite_view().width() // 2 - 5
        disparo_y = self.get_pos_y() + self.get_sprite_view().height()
        return Disparo(disparo_x, disparo_y, is_player=False)

    def destruir(self):
        """Esconde a nave quando destruída."""
        self._sprite_view.hide()